src/client.rs

```diff
@@ -157,10 +172,32 @@ impl DeezerClient {
         Ok(res.data)
     }
 
-    async fn get<T: DeserializeOwned>(&self, url: &str) -> Result<T> {
+    pub(crate) async fn get_subresource<T>(&self, id: u64, limit: Option<u32>,
+                                           offset: Option<u32>) -> Result<Vec<T>>
+        where
+            T: DeezerObject
+    {
+        let url = T::get_api_url(id);
+        let url = format!("{}/{}", BASE_URL, url);
+
+        let mut params: HashMap<String, String> = HashMap::new();
+        if let Some(_limit) = limit {
+            params.insert("limit".to_owned(), _limit.to_string());
+        }
+        if let Some(_offset) = offset {
+            params.insert("offset".to_owned(), _offset.to_string());
+        }
+
+        let res: DeezerArray<T> = self.get_with_params(&url, &params).await?;
+
+        Ok(res.data)
+    }
+
+    async fn get_with_params<T: DeserializeOwned>(&self, url: &str, query_params: &HashMap<String, String>) -> Result<T> {
```

Comment:
I would prefer if you could change this to an `Option<&HashMap<String, String>>` (maybe in another method so `get_with_params` keeps its signature) so we don't need `lazy_static` as an extra dependency.

Reply:
done :)